[baseline-alignment]Rename BaselineContext to BaselineAlignmentState.

sammygill opened the pull request:
#### 48c6a9eda4fdc93834a69222bd4db466c7b94e52
<pre>
[baseline-alignment]Rename BaselineContext to BaselineAlignmentState.
<a href="https://bugs.webkit.org/show_bug.cgi?id=257233">https://bugs.webkit.org/show_bug.cgi?id=257233</a>
rdar://109742609

Reviewed by Alan Baradlay.

This class is used primarily to provide an API for a formatting context
to interact with baseline sharing groups for baseline alignment. Also, a
baseline alignment context refers to a particular aspect of a formatting
context upon which baseline alignment occurs, such as lines in flex
layout.

Instead we can refer to this as BaselineAlignmentState since this is
the type of information we are holding in it.

* Source/WebCore/rendering/BaselineAlignment.cpp:
(WebCore::BaselineAlignmentState::BaselineAlignmentState):
(WebCore::BaselineAlignmentState::sharedGroup const):
(WebCore::BaselineAlignmentState::updateSharedGroup):
(WebCore::BaselineAlignmentState::findCompatibleSharedGroup):
(WebCore::BaselineContext::BaselineContext): Deleted.
(WebCore::BaselineContext::sharedGroup const): Deleted.
(WebCore::BaselineContext::updateSharedGroup): Deleted.
(WebCore::BaselineContext::findCompatibleSharedGroup): Deleted.
* Source/WebCore/rendering/BaselineAlignment.h:
* Source/WebCore/rendering/GridBaselineAlignment.cpp:
(WebCore::GridBaselineAlignment::updateBaselineAlignmentContext):
* Source/WebCore/rendering/GridBaselineAlignment.h:

Canonical link: <a href="https://commits.webkit.org/264484@main">https://commits.webkit.org/264484@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec13ec0c159ca8262b9715e81fd3fce25b914c6d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7766 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8039 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8219 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9408 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7920 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7772 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10029 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7959 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10783 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7902 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9013 "1 failures") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7127 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9519 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6321 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7067 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14735 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7472 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7193 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10603 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7690 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6269 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7017 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1858 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11229 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7421 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/GridBaselineAlignment.h:
```diff
@@ -74,12 +74,12 @@ class GridBaselineAlignment {
-    BaselineContextsMap m_rowAxisAlignmentContext;
-    BaselineContextsMap m_colAxisAlignmentContext;
+    BaselineAlignmentStatesMap m_rowAxisAlignmentContext;
+    BaselineAlignmentStatesMap m_colAxisAlignmentContext;
```
m_rowAxisAlignmentContext/m_colAxisAlignmentContext -> some other name?

sammygill replied:
Changed to `m_row/columnAxisBaselineAlignmentContexts`

Also changed `BaselineAlignmentStatesMap` to `BaselineAlignmentStateMap` since the map part of the name already implies multiple states

alanbaradlay replied:
> Changed to `m_row/columnAxisBaselineAlignmentContexts`
> 
> Also changed `BaselineAlignmentStatesMap` to `BaselineAlignmentStateMap` since the map part of the name already implies multiple states

does "context" postfix have some special meaning here that prevents you from going with "state" instead?

sammygill replied:
oh sorry I completely mistyped that last part! I ended up changing it to state

sammygill committed:
[baseline-alignment]Rename BaselineContext to BaselineAlignmentState.
https://bugs.webkit.org/show_bug.cgi?id=257233
rdar://109742609

Reviewed by Alan Baradlay.

This class is used primarily to provide an API for a formatting context
to interact with baseline sharing groups for baseline alignment. Also, a
baseline alignment context refers to a particular aspect of a formatting
context upon which baseline alignment occurs, such as lines in flex
layout.

Instead we can refer to this as BaselineAlignmentState since this is
the type of information we are holding in it.

* Source/WebCore/rendering/BaselineAlignment.cpp:
(WebCore::BaselineAlignmentState::BaselineAlignmentState):
(WebCore::BaselineAlignmentState::sharedGroup const):
(WebCore::BaselineAlignmentState::updateSharedGroup):
(WebCore::BaselineAlignmentState::findCompatibleSharedGroup):
(WebCore::BaselineContext::BaselineContext): Deleted.
(WebCore::BaselineContext::sharedGroup const): Deleted.
(WebCore::BaselineContext::updateSharedGroup): Deleted.
(WebCore::BaselineContext::findCompatibleSharedGroup): Deleted.
* Source/WebCore/rendering/BaselineAlignment.h:
* Source/WebCore/rendering/GridBaselineAlignment.cpp:
(WebCore::GridBaselineAlignment::updateBaselineAlignmentContext):
* Source/WebCore/rendering/GridBaselineAlignment.h:

Canonical link: https://commits.webkit.org/264484@main
